### DISCUSSION TITLE
How do we stack PCB's in pcb2blender module

### DISCUSSION THREAD

**Discussion by alextrical** [INTI-CMNB/KiBot#665] [Q&A] [answered]
Hi, all 

I'm working on a project that consists of 3 PCB's that will mate together with pin headers, I'm hoping to achieve something similar to this. ![](https://extensions.blender.org/media/thumbnails/86/8625a4b90f786079e4207fa1b3ad61b3beacc97924cb3fea4d60b3084943d75e_1920x1080.webp)

 Currently I believe I need to make use of the "Stack_boards" command, but I'm not sure how the KiCAD project has to be setup to make use of the command

![image-2](https://github.com/user-attachments/assets/487f06c8-e541-4067-b1fb-ccc9fec5039e)

If anyone can point me to some relevant reading materials or examples, it would be greatly appreciated

**Comment by set-soft** [comment, marked as answer]
Hi @alextrical !

This is defined by the `pcb2blender` tool. Here is the basic idea:
1. All the PCBs are in the same .kicad_pcb file
2. You need to create texts in your PCB that delimits its sub-pcb
3. The tool uses the coordinate for the text, so it doesn't matter its size or visibility
4. The top-left corner for a sub-board is marked using **PCB3D_TL_name**, where *name* is the name for the sub-PCB.
5. The bottom-right corner uses **PCB3D_BR_name**
6. Then you mark their relative coordinates  with texts like this **PCB3D_STACK_name1_ONTO_name2_distance** to indicate name1 is stacked onto name2 separated by a distance in the Z axis

Aligning the boards is tricky. If I'm right the point you mark with the **PCB3D_STACK_name1_ONTO_name2_distance** mark is just the top left corner for *name1* inside *name2*.

You can use the following config:

```yaml
kiplot:
  version: 1

import:
  - file: 3DRender_top
```
Note: this is *tests/yaml_samples/blender_export_template_1.kibot.yaml*

And the following [PCB](https://github.com/INTI-CMNB/KiBot/blob/dev/tests/board_samples/kicad_8/batteryPack.kicad_pcb)

You'll get:

![batteryPack-3D_blender_top](https://github.com/user-attachments/assets/53e409c9-62b8-4878-af93-56d6b4ecc9d4)

Which is what the file specifies:

![image](https://github.com/user-attachments/assets/962e3112-d6ff-4151-b388-1063e236702b)

**Reply by alextrical** [reply]
Sorry for the lack of reply, this was a really helpful answer :)
